src/sentry/notifications/manager.py

```diff
@@ -235,6 +235,26 @@ def get_for_users_by_parent(
             target__in=[user.actor.id for user in users],
         )
 
+    def get_for_team(
```

Comment:
I think you can name this something like `get_for_recipient_by_parent` so that it can take a `recipient: Union[Team, User]` and return only the parent-specific settings.